tests/python/examples/autodiff/test_regression.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+
+
+def mkdir_p(dir_path):
```

Comment:
Generally speaking, I don't think this kind of tool-like function is very appropriate here. But I didn't find matplotlib with a parameter like create_dir=true, so I put it here, and it might be duplicated in other UTs, any suggestions here?

Reply:
I think we can just place this here for now. In case it's used more widely, maybe put it in places like `tests/python/utils`

Reply:
yep, I will try to adjust it out